src/MoBi.Core/Domain/Model/MoBiSimulation.cs

```diff
@@ -73,9 +86,38 @@ public void RemoveAllChartTemplates()
 
       public bool IsCreatedBy(IBuildingBlock templateBuildingBlock)
       {
-         //TODO SIMULATION_CONFIGURATION
+         // We can consider the building block in-use if it belongs to a module that is in use.
+         if (templateBuildingBlock.Module != null)
+            return usesModuleBuildingBlock(templateBuildingBlock);
+
+         // Simple name match for building blocks that do not belong to a module
+         switch (templateBuildingBlock)
+         {
+            case IndividualBuildingBlock individualBuildingBlock:
+               return Equals(Configuration.Individual?.Name, individualBuildingBlock.Name);
+            case ExpressionProfileBuildingBlock expressionProfileBuildingBlock:
+               return Configuration.ExpressionProfiles.Any(x => Equals(x.Name, expressionProfileBuildingBlock.Name));
+         }
+
          return false;
-         // return MoBiBuildConfiguration.BuildingInfoForTemplate(templateBuildingBlock) != null;
+      }
+
+      private bool usesModuleBuildingBlock(IBuildingBlock templateBuildingBlock)
+      {
+         return BuildingBlocks.Any(buildingBlock => buildingBlock.Module.IsNamed(templateBuildingBlock.Module.Name) && buildingBlock.IsTemplateMatchFor(templateBuildingBlock));
+      }
+
+      public IReadOnlyList<IBuildingBlock> BuildingBlocks
```

Comment:
That looks like a property but underneath the hood, there is a lot of stuff going on with select many, concat etc.. Wonder if we should make it a function instead so that it's clear that there is something going on? Thoughts?